libyul/backends/evm/EVMDialect.cpp

```diff
@@ -190,6 +191,7 @@ EVMDialect const& EVMDialect::looseAssemblyForEVM(langutil::EVMVersion _version)
 EVMDialect const& EVMDialect::strictAssemblyForEVM(langutil::EVMVersion _version)
 {
 	static map<langutil::EVMVersion, unique_ptr<EVMDialect const>> dialects;
+	YulStringRepository::resetCallbacks().emplace_back([&] { dialects.clear(); });
```

Comment:
Why is this one !=?

Reply:
shoot!

Reply:
I'll try to make `YulStringRepository::resetCallbacks()` private.